packages/drivers/odsp-driver/src/retryUtils.ts

```diff
@@ -20,12 +20,24 @@ export async function runWithRetry<T>(
 ): Promise<T> {
     let retryAfter = 1000;
     const start = performance.now();
-    for (let retry = 1; ; retry++) {
+    let lastError: any;
+    for (let attempts = 1; ; attempts++) {
         if (checkDisposed !== undefined) {
             checkDisposed();
         }
         try {
-            return await api();
+            const result = await api();
+            if (attempts > 1) {
+                logger.sendTelemetryEvent(
+                    {
+                        eventName: "MultipleRetries",
+                        callName,
+                        attempts,
+                        duration: performance.now() - start,
```

Comment:
Just to make sure this is intentional, we're interested in the total time it took for the operation to succeed, not for how long it took this particular attempt, right?

Reply:
Yep, it is